### PR TITLE
IDE-1039 ECL IDE - Option to turn off UTF-BOM marker

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Signed-off-by: David de Hilster <david.dehilster@lexisnexisrisk.com>

### CODE REVIEW

File: comms/DiskAttribute.cpp [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #include "UnicodeFile.h"
 #include "md5.hpp"
 #include "ModuleHelper.h"
+#include "../wlib/PreferenceDlg.h"
```
Comment (GordonSmith) [inline comms/DiskAttribute.cpp:14]: Comms should not have a dependency on wlib

File: wlib/PreferenceDlg.cpp [hunk 2]
```diff
@@ -552,6 +553,9 @@ class CPrefEditorDlg : public CDialogImpl<CPrefEditorDlg>,
 		m_config = config;
 		m_TabWidth = m_config->Get(GLOBAL_TAB_WIDTH);
 		m_TabUseSpaces = m_config->Get(GLOBAL_TAB_USESPACES);
+		m_BOM = m_config->Get(GLOBAL_BOM);
```
Comment (GordonSmith) [inline wlib/PreferenceDlg.cpp:556]: You can supply a default value in the Get call (it should default to on), so no need for the `if` statement

File: wlib/PreferenceDlg.h [hunk 3]
```diff
@@ -23,6 +23,7 @@ static const TCHAR EditorSection[](_T("Editor"));
 static const SectionLabelDefault GLOBAL_MAINTAININDENT(SectionLabel(EditorSection, _T("MaintainIndent")), 1);
 static const SectionLabelDefault GLOBAL_TAB_WIDTH(SectionLabel(EditorSection, _T("TabWidth")), 2);
 static const SectionLabelDefault GLOBAL_TAB_USESPACES(SectionLabel(EditorSection, _T("UseSpaces")), true);
+static const SectionLabelDefault GLOBAL_BOM(SectionLabel(EditorSection, _T("BOM")), true);
```
Comment (GordonSmith) [inline wlib/PreferenceDlg.h:26]: This will need to be declared in comms and #include here,

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: wlib/PreferenceDlg.cpp [hunk 4]
```diff
@@ -552,6 +553,9 @@ class CPrefEditorDlg : public CDialogImpl<CPrefEditorDlg>,
...
+		if (m_BOM == -1)
+			m_BOM = 1;
```
[the remark this answers is not in the document]
Comment (dehilsterlexis) [reply wlib/PreferenceDlg.cpp:558]: It is needed to insure the default is set on. When this is the first time the config is looking at BOM and it is not there, it returns -1. Else zero or one.

[the remark this answers is not in the document]
Comment (GordonSmith) [reply wlib/PreferenceDlg.cpp:558]: You specified the default value here:  `static const SectionLabelDefault GLOBAL_BOM(SectionLabel(EditorSection, _T("BOM")), true);`?

File: comms/DiskAttribute.cpp (hunk 1 above)
Comment (dehilsterlexis) [reply comms/DiskAttribute.cpp]: No longer needed

File: wlib/PreferenceDlg.h (hunk 3 above)
Comment (dehilsterlexis) [reply wlib/PreferenceDlg.h]: Now in global.h

File: wlib/PreferenceDlg.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dehilsterlexis) [reply wlib/PreferenceDlg.cpp]: You are correct. Not needed but the defaults have to be integers for textboxes.

**Pull request merged**